server/src/com/cloud/api/ApiServer.java

```diff
@@ -540,7 +540,7 @@ public String handleRequest(final Map params, final String responseType, final S
                         throw new CloudRuntimeException("No APICommand annotation found for class " + cmdClass.getCanonicalName());
                     }
 
-                    BaseCmd cmdObj = (BaseCmd)cmdClass.newInstance();
+                    BaseCmd cmdObj = (BaseCmd)cmdClass.getConstructor(cmdClass).newInstance();
```

Comment:
@DaanHoogland what's the difference in doing `newInstance` directly, than getting the constructor first?

Reply:
it is deprecated in version 9.

Reply:
ok, stupifd mistake here; i added the class self to the list of parameters :(